### ISSUE TITLE
model training

### ISSUE THREAD

**Issue by 1215831011** [lovelyqian/CDFSOD-benchmark#2]
Hello, this is a very good work. I would like to know what needs to be done to train my own model based on your pre-trained model. I hope you can provide some references, thank you very much

**Comment by lovelyqian** [comment]
@1215831011  Hi, thanks for your question and the kind words. But What do you mean by training your own model, like different model architecture on the same target datasets? Or is it the same model as us but with different targets, or are both the model and the data different?

**Comment by 1215831011** [comment]
Thanks for your reply, I want to use the same model and train it with my own dataset based on your pre-trained model

**Comment by lovelyqian** [comment]
Then you can first organize your datasets e.g., "New" in the format as ours, then add this "New" dataset as another dataset, (could search the key work like "UODD" to see how to config it), and then just run the script. 

![image](https://github.com/user-attachments/assets/ac9564c1-850f-45aa-8f8e-00f2fe349b5a)

**Comment by 1215831011** [comment]
> 然后，您可以首先按照我们的格式组织您的数据集（例如“新”），然后将这个“新”数据集添加为另一个数据集（可以搜索“UODD”等关键工作来了解如何配置它），然后运行脚本。
> 
> ![图像](https://private-user-images.githubusercontent.com/49612387/351224509-ac9564c1-850f-45aa-8f8e-00f2fe349b5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MjEwODcsIm5iZiI6MTcyMTcyMDc4NywicGF0aCI6Ii80OTYxMjM4Ny8zNTEyMjQ1MDktYWM5NTY0YzEtODUwZi00NWFhLThmOGUtMDBmMmZlMzQ5YjVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDA3NDYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhMWM5ZWJjZTRjZTVkODVjNjg0NTQzZjIyZDZmOWJkMGJhY2JlMGFhNTg5YjkxZWZiYjUwZmU4Njk0MDhkOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FlCvsQ6I1EGjrS5RxwDAAi2hCmgpsIf956E_Ty_pkdQ)

Is it OK to use the data format of Coco, or does it need to be processed in other ways? Is it the same as DE-ViT's data processing method? I don't quite understand DE-ViT's data processing method

**Comment by lovelyqian** [comment]
> Is it OK to use the data format of Coco, or does it need to be processed in other ways? Is it the same as DE-ViT's data processing method? I don't quite understand DE-ViT's data processing method

The COCo format is okay. We follow the same format as detectron2

**Comment by 1215831011** [comment]
OK, thank you very much for your answer

**Comment by 1215831011** [comment]
Hello, thank you very much for your answer. My own dataset can already be trained. Can you provide a test demo of the UODD dataset?

**Comment by lovelyqian** [comment]
> Hello, thank you very much for your answer. My own dataset can already be trained. Can you provide a test demo of the UODD dataset?

Thanks for the feedback, for the numerical result, you can directly get it via running the bash main_results.sh, for the visualization, you can reference [draw_bounding_boxes](https://github.com/mlzxy/devit/blob/main/demo/demo.py#L59) function in [devit](https://github.com/mlzxy/devit/tree/main).

**Comment by lovelyqian** [comment]
for each class in the test set, randomly select one instance that has only 1 box in the image, and set it as the support image.

**Comment by wangqiudong** [comment]
> for each class in the test set, randomly select one instance that has only 1 box in the image, and set it as the support image.
Thanks for your greaT work. 
Most images in my dataset have 5~100 boxes, could I set them as the support images ?
and could I use them to build prototypes

**Comment by wangqiudong** [comment]
Thanks for your greaT work.
Most images in my dataset have 5~100 boxes, could I set them as the support images ?
and could I use them to build prototypes

**Comment by lovelyqian** [comment]
> Thanks for your greaT work. Most images in my dataset have 5~100 boxes, could I set them as the support images ? and could I use them to build prototypes

I think so, just make sure what is the task setting that you are working on. For example, for the few-shot OD setting, then should make sure the number of support instances is K-shot (depending on your task setting).  And yes, you could use the support images to build prototypes.
